tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -345,11 +345,20 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
 
     // Class definition
     if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
+      val (linkedInlinedInit, linkedMemberMethods) =
+        classEmitter.extractInlinedInit(linkedClass)(classCache)
+
+      // JS constructor
+      val inlinedInit = linkedInlinedInit.map { m =>
+        val inlinedInitCache = classCache.getInlinedInitCache()
+        inlinedInitCache.getOrElseUpdate(m.version,
+            classEmitter.genInlinedInit(className, m.tree)(inlinedInitCache))
+      }
```

Comment:
Not sure this is the best place to put all this. Have you considered putting all this inside `classEmitter` and simply filtering out the inlined constructor in the for loop below (the entire `emitter` knows about inlined constructors anyways, so it doesn't couple any tighter).

In any case, I do not see a use for the `inlinedInitCache` it gets cached right away afterwards...